### PR TITLE
Restructure README and add usage section

### PR DESCRIPTION
**Pull request by florisvdg**
I removed all installation instructions and refer to website docs instead.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,31 +16,24 @@ The SecretHub CLI provides the command-line interface to interact with SecretHub
 
 > [SecretHub][secrethub] is a developer tool to help you keep database passwords, API tokens, and other secrets out of IT automation scripts.
 
-## Installation
-
-### Download a binary distribution
+## Usage
```
Comment (florisvdg) [inline README.md]: I put `Usage` higher than `Installation`. You guys agree?

File: README.md [hunk 2]
```diff
@@ -16,31 +16,24 @@ The SecretHub CLI provides the command-line interface to interact with SecretHub
...
+db_user: myapp
+db_password: ${path/to/secret:latest}
 
-```sh
-mkdir -p /usr/local/secrethub
-tar -C /usr/local/secrethub -xzf secrethub-VERSION-OS-ARCH.tar.gz
+$ cat config.yml.tpl | secrethub inject
+db_user: myapp
+db_password: LEYkTdMCksCVMc4X3gpYN0fk
 ```
```
Comment (mackenbach) [inline README.md]: Add a line under the code example saying something like: `See the [getting started guide](...) for a more extensive introduction and check out the [CLI reference docs](...) for an overview of all commands.`

File: README.md [hunk 3]
```diff
@@ -16,31 +16,24 @@ The SecretHub CLI provides the command-line interface to interact with SecretHub
...
+## Installation
 
-Alternatively, you can add the folder containing the binary to your `PATH` environment variable:
-
-```sh
-export PATH=$PATH:/usr/local/secrethub/bin
-```
+The SecretHub CLI can be installed in various ways. Have a look at our [installation guide](https://secrethub.io/docs/getting-started/install) for more information.
 
 ### Build from source
```
Comment (mackenbach) [inline README.md]: We discussed moving this section to the `Development` section

File: README.md [hunk 4]
```diff
@@ -61,19 +54,14 @@ Now you can move it into the `PATH` to use it from any directory:
...
-## Test your installation
+### Test your installation
 
 Verify your installation works by running the following command:
 ```sh
 secrethub --version
 ```
 
-## Install auto-completion
+### Install auto-completion
```
Comment (mackenbach) [inline README.md]: I like your idea of moving this to the website and just having a reference to that in the README.

File: README.md [hunk 5]
```diff
@@ -16,31 +16,24 @@ The SecretHub CLI provides the command-line interface to interact with SecretHub
...
- ln -s /usr/local/secrethub/bin/secrethub /usr/local/bin/secrethub
- ```
+## Installation
 
-Alternatively, you can add the folder containing the binary to your `PATH` environment variable:
-
-```sh
-export PATH=$PATH:/usr/local/secrethub/bin
-```
+The SecretHub CLI can be installed in various ways. Have a look at our [installation guide](https://secrethub.io/docs/getting-started/install) for more information.
```
Comment (mackenbach) [inline README.md]: Consider refactoring this line to something like this:

--------------

Official distributions are available for the Linux, macOS, and Windows operating systems for both the 32-bit (`386`) and 64-bit (`amd64`) versions.

Check out the [installation guide](https://secrethub.io/docs/getting-started/install) for detailed instructions on how to install the SecretHub CLI on your platform of choice.

**Comment by SimonBarendse** [comment]
@florisvdg You can fix the CircleCI build by merging `develop` into your branch. This branch doesn't contain the configuration file yet.

**Pull request merged**